components/color_input.tsx

```diff
@@ -3,46 +3,50 @@
 
 import React from 'react';
 import {ChromePicker, ColorResult} from 'react-color';
-
-const hexRegex = /^#([\da-f]{3}|[\da-f]{6})$/i;
+import tinycolor from 'tinycolor2';
 
 type Props = {
     id: string;
-    color: string;
-    onChange?: (hex: string) => void;
+    onChange: (color: string) => void;
+    value: string;
 }
 
 type State = {
+    focused: boolean;
     isOpened: boolean;
-    hex: string;
+    value: string;
 }
 
-class ColorInput extends React.PureComponent<Props, State> {
+export default class ColorInput extends React.PureComponent<Props, State> {
     private colorPicker: React.RefObject<HTMLDivElement>;
     private colorInput: React.RefObject<HTMLInputElement>;
 
     public constructor(props: Props) {
         super(props);
         this.colorPicker = React.createRef();
         this.colorInput = React.createRef();
+
         this.state = {
+            focused: false,
             isOpened: false,
-            hex: this.props.color.toUpperCase(),
+            value: props.value,
         };
     }
 
-    public componentWillMount() {
-        this.setHex();
+    static getDerivedStateFromProps(props: Props, state: State) {
+        if (!state.focused && props.value !== state.value) {
```

Comment:
When dragging `<ChromePicker/>`, this currently nukes the value-change (appears frozen while dragging).

```suggestion
        if (!state.focused && !state.isOpened && props.value !== state.value) {
```

Reply:
This works fine for me. I'm seeing the textbox update live as you drag around
![image](https://user-images.githubusercontent.com/3277310/90152412-c973f580-dd55-11ea-944f-a71549bc2921.png)